compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -609,7 +612,14 @@ trait Applications extends Compatibility {
 
         case nil =>
           args match {
-            case arg :: args1 => fail(s"too many arguments for $methString", arg)
+            case arg :: args1 =>
+              val msg = arg match
+                case untpd.Tuple(Nil)
+                if applyKind == ApplyKind.Infix && funType.widen.isNullaryMethod =>
+                  i"can't supply unit value with infix notation because nullary method $methString takes no arguments; use dotted invocation instead: (...).${methRef.name}()"
```

Comment:
methString already contains the word "method"
```suggestion
                  i"can't supply unit value with infix notation because nullary $methString takes no arguments; use dotted invocation instead: (...).${methRef.name}()"
```